client/invalid_token_refresher_transport.go

```diff
@@ -0,0 +1,28 @@
+package client
+
+import (
+	"net/http"
+
+	"golang.org/x/oauth2"
+)
+
+type invalidTokenRefresherTransport struct {
+	base     *oauth2.Transport
+	tokenSrc oauth2.TokenSource
+}
+
+func (t *invalidTokenRefresherTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp, err := t.base.RoundTrip(req)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.StatusCode == http.StatusUnauthorized {
```

Comment:
What happens if this unauthorized response is not solved by re-requesting the token? It will get into an infinite cycle, right?
This might be a too broad condition

Reply:
Oh, I just realized that `t.base.RoundTrip(req)` will prevent that